internal/pass/connection/proxy_pool.go

```diff
@@ -0,0 +1,103 @@
+package connection
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/twofas/2fas-server/internal/common/logging"
+)
+
+// Proxy between Browser Extension and Mobile.
+type Proxy struct {
+	proxyPool *proxyPool
+	idLabel   string
+}
+
+func NewProxy(idLabel string) *Proxy {
+	proxyPool := &proxyPool{proxies: map[string]*proxyPair{}}
+	go func() {
+		ticker := time.NewTicker(30 * time.Second)
+		for {
+			<-ticker.C
+			proxyPool.deleteExpiresPairs()
+		}
+	}()
+	return &Proxy{
+		proxyPool: proxyPool,
+		idLabel:   idLabel,
+	}
+}
```

Comment:
this should be in proxy.go file, right?

Reply:
I reworked file structure here. I hope it's better now 😅